src/compositor/lipstickcompositor.cpp

```diff
@@ -732,3 +736,18 @@ void LipstickCompositor::readContent()
 {
     m_recorder->recordFrame(this);
 }
+
+void LipstickCompositor::surfaceCommitted()
+{
+    if (!isVisible() && !m_fakeRepaintTriggered) {
+        QTimer::singleShot(100, this, SLOT(fakeRepaint()));
```

Comment:
Not related to the logic itself, but I prefer to avoid signal/slot connections in graphics related matters as they are quite costly. a timerEvent() + QOjbect::start/killTimer is faster.